### PR TITLE
Fix for ScottishCovidResponse/simple_network_sim#501

### PR DESCRIPTION
**Pull request by widdowquinn**
These commits should:

1. bring `spec-file-macos.txt` up to date with the Linux spec file
2. add a fixture that removes the `data_pipeline_tests/output` directory (created during testing) at the end of the `pytest` session.

I believe these address ScottishCovidResponse/SCRCIssueTracking#501, though I'm not set on the fixture name, or even the approach - but I think the `pytest` `tmp_path` fixture is not feasible as the data locations are defined in `config.yaml` for working with `Datastore()`.

Also, I only found two stray output files on a local test run, though the issue title suggested there would be more. Did I miss some?

### CODE REVIEW

**Comment by bobturneruk** [comment]
See comments on https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/501. We should consider whether having a spec-file for OSX or a cross-platform environment.yml  is best, see https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/545.

**Comment by bobturneruk** [comment]
Resolves https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/441

File: tests/conftest.py [hunk 1]
```diff
@@ -29,3 +30,14 @@ def base_data_dir():
...
+
+@pytest.fixture(autouse=True, scope="session")
+def teardown_remove_data():
+    """Remove test output created during testing.
+
+    Datasets defined in data_pipeline_inputs/config.yaml can't be handled with
+    pytest's tmp_path, so are cleaned up here.
+    """
+    yield
+    shutil.rmtree(FIXTURE_DIR / "data_pipeline_inputs" / "output")
```
Comment (aflag) [inline tests/conftest.py:43]: I think the access log is stored in `FIXTURE_DIR / "data_pipeline_inputs"`: `access*.yaml`. If we could delete those too it'd be nice.

**Comment by widdowquinn** [comment]
BTW, do you want to respect or ignore the `pydocstyle` D213 warning that Codacy is flagging up? I could go with either D212 or D213's format ([error codes](https://pep257.readthedocs.io/en/latest/error_codes.html))

**Comment by bobturneruk** [comment]
Please ignore D212 and D213 - my understanding is these should not normally be checked for - I will tell codacy. Please also ignore codacy's rude comment "Not up to standards."!

**Comment by widdowquinn** [comment]
> Please also ignore codacy's rude comment "Not up to standards."!

I normally do ;)

File: tests/conftest.py (hunk 1 above)
Comment (widdowquinn) [reply tests/conftest.py:43]: > I think the access log is stored in `FIXTURE_DIR / "data_pipeline_inputs"`: `access*.yaml`. If we could delete those too it'd be nice.

`config.yaml` indicates an `access.log` file might be created, but I've yet to see one appear with local testing. I'm happy to add this speculatively, though.

Comment (aflag) [reply tests/conftest.py:44]: I think they are supposed to be called `access-*.yaml`. It's odd that they are not showing up there. Maybe they are getting created somewhere else?

**Pull request merged**